services/app-web/src/pages/SubmissionSummary/UnlockModal.tsx

```diff
@@ -0,0 +1,173 @@
+    const [focusErrorsInModal, setFocusErrorsInModal] = useState(true) // when formik errors occur
+    const modalFormInitialValues = {
+        unlockReason: '',
+    }
+
+    const [unlockHealthPlanPackage] = useUnlockHealthPlanPackageMutation()
+    const formik = useFormik({
+        initialValues: modalFormInitialValues,
+        validationSchema: Yup.object().shape({
+            unlockReason: Yup.string().defined(
+                'Reason for unlocking submission is required'
+            ),
+        }),
+        onSubmit: (values) => onModalSubmit(values),
+    })
+
+    // Focus unlockReason field in the unlock modal on submit click when errors exist
+    useEffect(() => {
+        if (focusErrorsInModal && formik.errors.unlockReason) {
+            const fieldElement: HTMLElement | null = document.querySelector(
+                `[name="unlockReason"]`
+            )
+
+            if (fieldElement) {
+                fieldElement.focus()
+                setFocusErrorsInModal(false)
+            } else {
+                console.log('Attempting to focus element that does not exist')
+            }
+        }
+    }, [focusErrorsInModal, formik.errors])
+
+    const onModalSubmit = async (values: typeof modalFormInitialValues) => {
+        const { unlockReason } = values
+        await onUnlock(unlockReason)
+    }
+
+    const onUnlock = async (unlockReason: string) => {
+        const result = await unlockMutationWrapper(
+            unlockHealthPlanPackage,
+            healthPlanPackage.id,
+            unlockReason
+        )
+
+        if (result instanceof Error) {
+            console.error(
+                'ERROR: got an Apollo Client Error attempting to unlock',
+                result
+            )
+            setModalAlert('Error attempting to unlock. Please try again.')
+            modalRef.current?.toggleModal(undefined, false)
+        } else if (isGraphQLErrors(result)) {
+            console.error('ERROR: got a GraphQL error response', result)
+            if (result[0].extensions.code === 'BAD_USER_INPUT') {
+                setModalAlert(
+                    'Submission is already unlocked. Please refresh and try again.'
+                )
+            } else {
+                setModalAlert('Error attempting to unlock. Please try again.')
+            }
+            modalRef.current?.toggleModal(undefined, false)
+        } else {
+            const unlockedSub: HealthPlanPackage = result
+            modalRef.current?.toggleModal(undefined, false)
+            console.log('Submission Unlocked', unlockedSub)
+        }
+    }
+
+    return (
+        <Modal
+            modalHeading="Reason for unlocking submission"
+            id="unlockReason"
+            onSubmit={() => {
+                setFocusErrorsInModal(true)
+                formik.handleSubmit()
+            }}
+            isSubmitting={formik.isSubmitting}
```

Comment:
This is the only related code change, everything else is copy paste from submission summary